installers/win_installer.py

```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2020 Spyder Project Contributors.
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+#
+# Copyright © 2018 Nicholas H.Tollervey.
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Script to create a windows installer using pynsist.
+
+Based on the Mu's win_installer.py script
+https://github.com/mu-editor/mu/blob/master/win_installer.py
+"""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+
+import yarg
+
+
+# Packages to remove from the requirements for example pip
+
+UNWANTED_PACKAGES = ['pip']
+
+# Packages to skip checking for wheels and instead add them directly in the
+# 'packages' section
+
+SKIP_PACKAGES = ['bcrypt']
+
+# The pynsist requirement spec that will be used to install pynsist in
+# the temporary packaging virtual environment.
+
+PYNSIST_REQ = "pynsist==2.5.1"
+
+# The pynsist configuration file template that will be used. Of note,
+# with regards to pynsist dependency collection and preparation:
+# - {pypi_wheels} will be downloaded by pynsist from PyPI.
+# - {packages} will be copied by pynsist from the current Python env.
+
+PYNSIST_CFG_TEMPLATE = """
+[Application]
+name={name}
+version={version}
+entry_point={entrypoint}
+icon={icon_file}
+publisher={publisher}
+license_file={license_file}
+[Python]
+version={python_version}
+bitness={bitness}
+format=bundled
+[Include]
+pypi_wheels=
+    {pypi_wheels}
+packages=
+    tkinter
+    _tkinter
+    turtle
+    win32api
+    win32security
+    ntsecuritycon
+    {packages}
+[Build]
+installer_name={installer_name}
+"""
+
+
+def subprocess_run(args):
+    """
+    Wrapper-function around subprocess.run.
+
+    When the sub-process exits with a non-zero return code,
+    prints out a message and exits with the same code.
+    """
+    cp = subprocess.run(args)
+    try:
+        cp.check_returncode()
+    except subprocess.CalledProcessError as exc:
+        print(exc)
+        sys.exit(cp.returncode)
+
+
+def create_packaging_env(
+        target_directory, python_version, name="packaging-env",
+        conda_path=None):
+    """
+    Create a Python virtual environment in the target_directry.
```

Comment:
```suggestion
    Create a Python virtual environment in the target_directory.
```

Minor spelling error.